src/Sulu/Bundle/CustomUrlBundle/EventListener/UpdateLocalizationListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CustomUrlBundle\EventListener;
+
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Update localization in case of custom-url route.
+ */
+class UpdateLocalizationListener
+{
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    public function __construct(RequestAnalyzerInterface $requestAnalyzer)
+    {
+        $this->requestAnalyzer = $requestAnalyzer;
+    }
+
+    /**
+     * Update locale in request analyzer.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
```

Comment:
I know this is done in many places, but @webmozart indicated that it would be better to use more domain specific method names here. So I would call this one `updateRequestAnalyzerLocalization`.